Enable auth config on Milvus resource

timheuer opened the pull request:
- Updates Milvus hosting to leverage dynamically generated default password if none provided (similar to other resources) and resulting cascading changes + tests.
- Updates tags for contianer image that support setting auth

## Description

Milvus added the ability to set the root password. This change adapts to that and follows similar pattern to other config-based auth (e.g., MySql is a good analog). This requires the support of a minimum new Milvus version (2.4.7). 

Fixes #4966

## Checklist

- Is this feature complete?
  - [ ] Yes. Ready to ship.
  - [x] No. Follow-up changes expected. - haven't seen a pattern where we've made a change that requires a specific new resource version
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
  - [ ] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [x] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [x] No - follows same pattern as existing pattern/resources
  - [ ] No
- Does the change require an update in our Aspire docs?
  - [x] Yes
    - Link to aspire-docs issue: [dotnet/docs-aspire#1475](https://github.com/dotnet/docs-aspire/issues/1475)
  - [ ] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5229)

timheuer committed:
Fix #4966

- Updates Milvus hosting to leverage dynamically generated default password if none provided (similar to other resources) and resulting cascading changes + tests.
- Updates tags for contianer image that support setting auth

timheuer committed:
Fix #4966

- Updates Milvus hosting to leverage dynamically generated default password if none provided (similar to other resources) and resulting cascading changes + tests.
- Updates tags for contianer image that support setting auth

eerhardt commented on src/Aspire.Hosting.Milvus/MilvusBuilderExtensions.cs:
```diff
@@ -41,13 +41,13 @@ public static class MilvusBuilderExtensions
     {
-        ArgumentNullException.ThrowIfNull(apiKey, nameof(apiKey));
+        var apiKeyParameter = apiKey?.Resource ??
+            ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-Key", special: false);
```
Why `special: false`?

eerhardt replied:
(super nit): in other places we use lower case names: `$"{name}-password"` but here we have PascalCase `-Key`. Should we be consistent?

eerhardt commented on tests/Aspire.Hosting.Tests/Milvus/AddMilvusTests.cs:
```diff
@@ -89,7 +89,7 @@ public async Task MilvusCreatesConnectionString()
 
         var connectionString = await connectionStringResource.GetConnectionStringAsync();
-        Assert.Equal($"Endpoint=http://localhost:19530;Key=pass", connectionString);
+        Assert.Equal($"Endpoint=http://localhost:19530;Key=root:pass", connectionString);
```
These tests have all moved out of `Aspire.Hosting.Tests` and are in `Aspire.Hosting.Milvus.Tests` now. Can you sync main?

timheuer replied:
doh, lots of changes, forgot to rebase...I will do that and resubmit I think. -- keep feedback coming and i'll address.

joperezr committed:
Update dependency versions (#5120)

* Update dependency versions

* Revert MongoDB.Driver update

adamint committed:
Add support for all cultures corresponding to localizations we have (#5143)

* Add support for all cultures corresponding to localizations we have

* add unit test

* Clean up per pr comments

radical committed:
Aspire.Hosting.Testing: Improve error message when loading (#5218)

* Aspire.Hosting.Testing: Improve error message when loading

.. launchsettings fails. This matches the code in
`DistributedApplicationFactory.GetLaunchSettings`.

* [playground] Fix CosmosEndToEnd.AppHost's launchsettings.json - trailing comma

drewnoakes committed:
Use unique ID for each Plotly chart in the DOM (#5224)

Enables the ability to have multiple charts on screen at once.

radical committed:
[tests] Move more tests to helix by overriding dcp path from assembly metadata (#5215)

`AppHost` build embeds metadata with paths for `DcpCliPath`, `DcpExtensionsPath`, and `DcpBinPath`. This ties the apphost to the machine that it was built on.
This PR:
- When `DcpPublisher:CliPath` is available from the configuration, set the other two paths based on that. 
- Makes `$(RunTestsOnHelix)=true` the default.

This allows running most of the tests on helix. The remaining ones depend on being able to run the test projects which doesn't work on helix because the path to the project files is embedded in the assembly metadata.

Also, download helix test result `.trx` files to have all the `.trx` files available in `artifacts`.

Contributes to https://github.com/dotnet/aspire/issues/2036 .

Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

cmdkeen committed:
Oracle OpenTelemetry support (#4078)

* Bumping versions and adding Oracle.ManagedDataAccess.OpenTelemetry

* Adding default OTel configuration

* Tracing setting from the oracle-samples

* Playground to show reading from DB

* Tracing conformance test

* Mention of tracing in Readme

* Reverting NuGet.config change

* Revert NuGet.config

* WIP on tests - TestContainer issues

* All tests passing

* Actually works with base Testcontainers.Oracle

* Unchanging NuGet.config again

* retrigger checks

* More JSON validation
Fail if can't connect to DB to help debug test issue

* Reverting back to custom builder

* Adding RunTestsOnHelix

* Back to TestContainers

* Apply suggestions from code review

Co-authored-by: Ankit Jain <radical@gmail.com>

* Waiting for PDB to be opened

* Removing wrong Aspire Manifest
Long sleep to try and wait for Oracle

* Waiting for database open confirmation

* CollectionFixture to avoid spinning up 3 Oracle instances and timeouts

* Updating to RequiresDockerAttribute

* Reverting to default tracing options

* Standardising on port 1521 and waiting until SQL execution succeeds

* Trying Postgres port for for Helix runner

* Logging test container logs to see why not starting

* Generated aspire-manifest

* Not depending on Postrges DbContext

* Removing CanConnect to get specific ORA exception details

* Specifying hostname & back to random port
Tweaking wait log message

* Backing out changes for detailed logging

* Tests for settings to AddOracleDatabaseDbContext

* Updating unshipped API

* Reverting to default EnableConnectionLevelAttributes

* Updating ODP.Net to latest version

* PR feedback

---------

Co-authored-by: Mitch Denny <midenn@microsoft.com>
Co-authored-by: Sébastien Ros <sebastienros@gmail.com>
Co-authored-by: Ankit Jain <radical@gmail.com>
Co-authored-by: Eric Erhardt <eric.erhardt@microsoft.com>

Alirexaa committed:
Extract Aspire.Hosting.SqlServer.Tests project (#5056)

timheuer committed:
More fixes after main sync
